### PR TITLE
Add embed handler for gfycat.

### PR DESCRIPTION
**Pull request by miina**

### CODE REVIEW

File: tests/test-class-amp-gfycat-embed-handler.php [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	/**
+	 * Test conversion.
+	 *
+	 * @param string $source Source.
+	 * @param string $expected Expected.
+	 * @dataProvider get_conversion_data
+	 */
+	public function test__conversion( $source, $expected ) {
+		if ( ( version_compare( get_bloginfo( 'version' ), '4.9', '<' ) ) ) {
+			$this->markTestSkipped( 'Gfycat is not supported in this WP version.' );
```
[the remark this answers is not in the document]
Comment (kienstra) [reply tests/test-class-amp-gfycat-embed-handler.php]: Hi @miina, good question. 

Could you try adding [this part](https://github.com/Automattic/amp-wp/blob/ac37c380c5cf265c3e3f1ef6865b62db80bab6fa/tests/test-amp-soundcloud-embed.php#L30-L47) of the `setUp` method from `AMP_SoundCloud_Embed_Test` to `AMP_Gfycat_Embed_Test`?

That worked in testing PR #1138 that I opened, even without `$this->markTestSkipped()`.

The [failed build](https://travis-ci.org/Automattic/amp-wp/jobs/376843147) that you linked to used WordPress version `4.7`. When I set my local to that version, the test here failed (without any modification). Strangely, the code works fine on the front-end. Just not in the tests in that case.

**Comment by miina** [comment]
@kienstra Thanks for reviewing and taking the time to create a separate PR for the tests fix! The issues should be addressed now.

**Comment by kienstra** [comment]
**Request To Merge**

Hi @westonruter,
When you have a chance, could you please merge this pull request? @miina's work here looks good.

**Comment by westonruter** [comment]
I'm curious: the amp-iframe and the amp-gfycat components seemed to render exactly the same. Why is the amp-gfycat component even necessary?

**Comment by miina** [comment]
@westonruter This is an example output with `amp-gfycat` (with URL `http://gfycat.com/gifs/detail/webbedbossycollardlizard`, as you mentioned in comments there's an issue with the logic currently that expects the URL to be in format `...gfycat.com/gifs/detail/[ID]`, should allow `gfycat.com/[ID]`, too.):
screen shot 2018-05-11 at 8 26 00 pm

And this without:
screen shot 2018-05-11 at 8 26 54 pm

Are you thinking that `amp-gfycat` isn't necessary at all since `amp-iframe` is already generated for gfycat?

**Comment by westonruter** [comment]
> Are you thinking that amp-gfycat isn't necessary at all since amp-iframe is already generated for gfycat and since there might be no difference for the user?

Right. I'm confused as to why the `amp-gfycat` component exists when an `amp-iframe` seems to work just as well? The main thing that it does is allow you to easily provide just a `gfyid` to construct the embed. Additionally, there is an `noautoplay` attribute which can be used but it merely adds an `? autoplay=0` to the iframe'd URL:

https://github.com/ampproject/amphtml/blob/04f3484ee0a4506a4cd205b373d7be3763d02649/extensions/amp-gfycat/0.1/amp-gfycat.js#L132-L136

That `noautoplay` attribute seems to be the main benefit here from a WordPress context, since gfycat isn't forwarding along the `autoplay` query parameter to the oEmbed response. But if not using oEmbed, then it could just be constructed with `amp-iframe`.

**Comment by westonruter** [comment]
Correction, it looks like the automatic addition of the placeholder is more of a key aspect of the component: https://github.com/ampproject/amphtml/blob/04f3484ee0a4506a4cd205b373d7be3763d02649/extensions/amp-gfycat/0.1/amp-gfycat.js#L84-L108

So yeah, I think we should go ahead with the custom handler you've worked on here. But it just needs to be finished off.

File: includes/embeds/class-amp-gfycat-embed-handler.php [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+			}
+			if ( preg_match( '/height=["\']?(\d+)/', $return, $matches ) ) {
+				$attr['height'] = $matches[1];
+			}
+
+			if ( empty( $attr['height'] ) ) {
+				return $return;
+			}
+
+			$layout = 'intrinsic';
```
Comment (westonruter) [inline includes/embeds/class-amp-gfycat-embed-handler.php]: This causes an AMP validation error:

> The specified layout 'INTRINSIC' is not supported by tag 'amp-gfycat'.

**Pull request merged**